pkg/k8s/allowlist.go

```diff
@@ -18,223 +18,237 @@ import (
+		"IAMAccessBoundaryPolicy",
+		"IAMAuditConfig",
+		"IAMCustomRole",
+		"IAMPartialPolicy",
+		"IAMPolicy",
+		"IAMPolicyMember",
+		"IAMServiceAccount",
+		"IAMServiceAccountKey",
+		"IAMWorkforcePool",
+		"IAMWorkforcePoolProvider",
+		"IAMWorkloadIdentityPool",
+		"IAMWorkloadIdentityPoolProvider",
+		"IAPBrand",
+		"IAPIdentityAwareProxyClient",
+		"IdentityPlatformConfig",
+		"IdentityPlatformOAuthIDPConfig",
+		"IdentityPlatformTenant",
+		"IdentityPlatformTenantOAuthIDPConfig",
+		"KMSCryptoKey",
+		"KMSKeyRing",
+		"LoggingLogBucket",
+		"LoggingLogExclusion",
+		"LoggingLogMetric",
+		"LoggingLogSink",
+		"LoggingLogView",
+		"MemcacheInstance",
+		"MonitoringAlertPolicy",
+		"MonitoringDashboard",
+		"MonitoringGroup",
+		"MonitoringMetricDescriptor",
+		"MonitoringMonitoredProject",
+		"MonitoringNotificationChannel",
+		"MonitoringService",
+		"MonitoringServiceLevelObjective",
+		"MonitoringUptimeCheckConfig",
+		"NetworkConnectivityHub",
+		"NetworkConnectivitySpoke",
+		"NetworkSecurityAuthorizationPolicy",
+		"NetworkSecurityClientTLSPolicy",
+		"NetworkSecurityServerTLSPolicy",
+		"NetworkServicesEndpointPolicy",
+		"NetworkServicesGRPCRoute",
+		"NetworkServicesGateway",
+		"NetworkServicesHTTPRoute",
+		"NetworkServicesMesh",
+		"NetworkServicesTCPRoute",
+		"NetworkServicesTLSRoute",
+		"OSConfigGuestPolicy",
+		"OSConfigOSPolicyAssignment",
+		"PrivateCACAPool",
+		"PrivateCACertificate",
+		"PrivateCACertificateAuthority",
+		"PrivateCACertificateTemplate",
+		"Project",
+		"PubSubLiteReservation",
+		"PubSubSchema",
+		"PubSubSubscription",
+		"PubSubTopic",
+		"RecaptchaEnterpriseKey",
+		"RedisInstance",
+		"ResourceManagerLien",
+		"ResourceManagerPolicy",
+		"RunJob",
+		"RunService",
+		"SQLDatabase",
+		"SQLInstance",
+		"SQLSSLCert",
+		"SQLUser",
+		"SecretManagerSecret",
+		"SecretManagerSecretVersion",
+		"Service",
+		"ServiceDirectoryEndpoint",
+		"ServiceDirectoryNamespace",
+		"ServiceDirectoryService",
+		"ServiceIdentity",
+		"ServiceNetworkingConnection",
+		"SourceRepoRepository",
+		"SpannerDatabase",
+		"SpannerInstance",
+		"StorageBucket",
+		"StorageBucketAccessControl",
+		"StorageDefaultObjectAccessControl",
+		"StorageNotification",
+		"StorageTransferJob",
+		"TagsTagBinding",
+		"TagsTagKey",
+		"TagsTagValue",
+		"VPCAccessConnector",
+	}
+
+	// v1beta1KindsWithComputedFieldsUnderStatus contains all the existing
+	// v1beta1 kinds that have computed fields directly under 'status' in the
+	// schema. This list is the same as
+	// v1beta1KindsWithStateIntoSpecMergeSupport.
+	// Any newly supported v1beta1 kinds should NOT have computed fields
+	// directly under 'status' in the schema.
+	v1beta1KindsWithComputedFieldsUnderStatus = v1beta1KindsWithStateIntoSpecMergeSupport
+)
 
 func SupportsStateIntoSpecMergeInKind(kind string) bool {
 	return isValueInAllowlist(kind, v1beta1KindsWithStateIntoSpecMergeSupport)
 }
 
+func SupportsComputedFieldsUnderStatusInKind(kind string) bool {
```

Comment:
IIUC callers should call `HasComputedFieldsUnderObservedState` instead ... Make this private or inline?

Reply:
Done.